force-app/main/default/classes/BDI_ProcessSelectedDIs_CTRL.cls

```diff
@@ -0,0 +1,113 @@
+/**
+    Copyright (c) 2022, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2022
+* @description Controller for visualforce page enabling a selected set of Data Import records 
+* to be processed via list view button
+*/
+public with sharing class BDI_ProcessSelectedDIs_CTRL {
+
+    private String diIds = '';
+    PageReference listView = new ApexPages.Action('{!List}').invoke();
+    private Integer maxNumOfDi = 100;
+    public Boolean error {get; set;}
+
+    /*********************************************************************************************************
+    * @description Constructor that extracts the Data Import Ids from the URL and performs some checks on them
+    * before reformatting the ids for the redirect.
+    */
+    public BDI_ProcessSelectedDIs_CTRL(ApexPages.StandardSetController controller) {
+
+        DataImport__c[] diRecs = (List<DataImport__c>)controller.getSelected();
```

Comment:
Can we break up the responsibilities of this constructor, we seem to have 2 distinct responsibilities here breaking it up which will help us unit test the different behaviours.